tests/ctst/HOW_TO_WRITE_TESTS.md

```diff
@@ -0,0 +1,121 @@
+# How to write a test that runs in parallel
+
+Writting intergation / behavior tests that run in parallel is not trivial.
+This page lists all the rules to follow when writing a test that runs in
+parallel with other tests.
+
+*This is specific to Zenko, but the rules are applicable for any suite of*
+*tests in parallel.*
+
+**Rule #1**
+
+> Tests must be idempotent.
+
+In parallel, the tests can run in any order. It's not because a test works at a
+given time, that it will work if the test order changes after adding new ones
+or renaming files. You must ensure the scenarios are fully idempotent.
+"Flakies" are mostly due to non-idempotent tests, then bugs. It is rarely due
+to the platform.
+
+**Rule #2**
+
+> Use unique resources names. S3 Bucket, Account and S3 Objects should be
+> unique.
+
+It limits the risks of collision between tests.
+For cross-account scenarios, use unique S3 Policies and S3 Roles.
+
+If data is used, such as lock files, ensure they are named appropriately.
+
+**Rule #3**
+
+> Avoid reconfiguring the environment during test execution.
+
+If a reconfiguration triggers a reconciliation of the resources, running tests
+might be affected. Also, it restarts the services, which leads to missing logs,
+harder debugging, and longer test execution.
+
+Examples:
+
+- Location creation;
+- Bucket Website endpoints.
+
+... must be created before the test start, or in a dedicated set of test.
```

Comment:
> or in a dedicated set of test

not clear what you mean by "set of test" here : a separate test suite, a separate execution of ctst , ... ?